tree/ntuple/v7/doc/specifications.md

```diff
@@ -1,159 +1,560 @@
+```
+
+_Envelope version_: Envelope types are versioned independently from each other.
+The envelope version is the version that the writer used to build the envelope.
+
+_Minimum version_:
+A reader must support at least this version in order to extract meaningful data from the envelope.
+If the envelope version is larger than the minimum version, there might be additional data in the envelope
+that older readers can safely ignore.
+
+_CRC32_: Checksum of the envelope and the payload
+
+Note that the size of envelopes is given by the RNTuple anchor (header, footer)
+or by a locator that references the envelope.
+
+
+### Header Envelope
+
+The header consists of the following elements:
+
+ - Feature flag
+ - String: name of the ntuple
+ - String: description of the ntuple
+ - String: identifier of the library or program that writes the data
+ - List frame: list of field record frames
+ - List frame: list of column record frames
+ - List frame: list of alias column record frames
+ - List frame: list of extra type information
+
+#### Field Description
+
+Every field record frame of the list of fields has the following contents
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         Field Version                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          Type Version                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
++                        Parent Field ID                        +
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|        Structural Role        |             Flags             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+The field version and type version are used for schema evolution.
+
+If the flag 0x01 (_repetitive field_) is set, the field represents a fixed sized array.
+In this case, an additional 64bit integer specifies the size of the array.
+
+The block of integers is followed by a list of strings:
+
+- String: field name
+- String: type name
+- String: type alias
+- String: field description
+
+The order of fields matters: every field gets an implicit field ID
+which is equal the zero-based index of the field in the serialized list;
+subfields are ordered from smaller IDs to larger IDs.
+Top-level fields have their own field ID set as parent ID.
+
+The flags field can have one of the following bits set
+
+| Bit      | Meaning                                                                    |
+|----------|----------------------------------------------------------------------------|
+| 0x01     | Repetitive field, i.e. for every entry $n$ copies of the field are stored  |
+| 0x02     | Alias field, the columns referring to this field are alias columns         |
```

Comment:
I am confused ... is the field itself an alias which refers to other columns or is it an "alias target"?